### PR TITLE
DM-27508 Allow collection name to be specified in DBB BM Gen3 Ingester config

### PR DESCRIPTION
**Pull request by mxk62**

### CODE REVIEW

File: doc/lsst.dbb.buffmngrs.endpoint/get-started.rst [hunk 1]
```diff
@@ -79,8 +79,8 @@ component that you may adapt to your local configuration:
    database:
      engine: "postgres://tester:password@localhost:5432/test"
      tablenames:
-       file: files
-       event: gen2_file_events
+       file:
```
Comment (MichelleGower) [inline doc/lsst.dbb.buffmngrs.endpoint/get-started.rst:82]: Documentation for schema key in the etc/*.yaml files?

File: python/lsst/dbb/buffmngrs/endpoint/ingester/plugins.py [hunk 2]
```diff
@@ -127,62 +144,69 @@ def execute(self, filename):
 
 
 class Gen3Ingest(Plugin):
-    """Ingest a file to Gen2 Butler repository.
+    """Ingest a file to Gen3 Butler repository.
+
+    The LSST Gen3 Middleware presumably supports parallel ingestion of
+    multiple files.  However, it does not expose information which files
+    were ingested successfully and which failed to ingest.  Hence,
+    the plugin is deliberately restricted to ingesting one file at the time.
```
Comment (MichelleGower) [inline python/lsst/dbb/buffmngrs/endpoint/ingester/plugins.py]: Parallelism is instead done at the ? level by setting ?.

File: python/lsst/dbb/buffmngrs/endpoint/ingester/plugins.py [hunk 3]
```diff
@@ -127,62 +144,69 @@ def execute(self, filename):
...
+        class_ = doImport(self.config["ingest_task"])
+        config = class_.ConfigClass()
+        config.transfer = self.config["transfer"]
+        config_overrides = ConfigOverrides()
+        if self.config["config_file"] is not None:
+            config_overrides.addFileOverride(self.config["config_file"])
+        if self.config["config"] is not None:
+            for key, val in config.items():
+                config_overrides.addValueOverride(key, val)
+        self.task = class_(config=config, butler=butler)
```
Comment (pehartma) [inline python/lsst/dbb/buffmngrs/endpoint/ingester/plugins.py]: Needs "config_overrides.applyTo(config)"

Comment (mxk62) [reply python/lsst/dbb/buffmngrs/endpoint/ingester/plugins.py]: Indeed! Thanks for catching it.

File: python/lsst/dbb/buffmngrs/endpoint/ingester/commands.py [hunk 4]
```diff
@@ -86,37 +86,48 @@ def start(filename, dump, validate):
...
-        available = set(inspect(engine).get_table_names())
-    except Exception as ex:
-        msg = f"{ex}"
-        logger.error(msg)
-        raise RuntimeError(msg)
-    else:
-        missing = required - available
-        if missing:
-            msg = f"table(s) {', '.join(missing)} not found in the database"
+    required = set()
```
Comment (MichelleGower) [inline python/lsst/dbb/buffmngrs/endpoint/ingester/commands.py]: Are these new lines (~92-117) the same as in the finder/commands.py?  If so, think about putting them in a separate function.

File: python/lsst/dbb/buffmngrs/endpoint/ingester/plugins.py [hunk 5]
```diff
@@ -20,10 +20,12 @@
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 import importlib
```
Comment (MichelleGower) [inline python/lsst/dbb/buffmngrs/endpoint/ingester/plugins.py]: importlib no longer used?

**Review by MichelleGower — APPROVED** [review]
Some minor comments.  After changes, good to merge.

File: python/lsst/dbb/buffmngrs/endpoint/ingester/plugins.py (hunk 2 above)
Comment (mxk62) [reply python/lsst/dbb/buffmngrs/endpoint/ingester/plugins.py]: I'll add

    To enable concurrent ingests, use Ingester's ``num_threads`` setting  instead.

to hint how to enable concurrent ingests in the manager.

File: python/lsst/dbb/buffmngrs/endpoint/ingester/commands.py (hunk 4 above)
Comment (mxk62) [reply python/lsst/dbb/buffmngrs/endpoint/ingester/commands.py]: As suggested, I rewrote the repeated code as functions and put them in `utils.py` so both the Finder and the Ingester can use them.

**Review by pehartma — APPROVED** [review]

**Pull request merged**